frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -558,11 +561,15 @@ end
 function ReaderRolling:onGotoNextChapter()
     local visible_page_count = self.ui.document:getVisiblePageCount()
     local pageno = self.current_page + (visible_page_count > 1 and 1 or 0)
-    -- Find next chapter start
-    local new_page = self.ui.document:getNextPage(pageno)
-    while new_page > 0 do
-        if self.ui.toc:isChapterStart(new_page) then break end
-        new_page = self.ui.document:getNextPage(new_page)
+    if self.ui.document:hasHiddenFlows() then
+        -- Find next chapter start
+        local new_page = self.ui.document:getNextPage(pageno)
+        while new_page > 0 do
+            if self.ui.toc:isChapterStart(new_page) then break end
+            new_page = self.ui.document:getNextPage(new_page)
+        end
+    else
+        local new_page = self.ui.toc:getNextChapter(pageno) or 0
     end
     if new_page > 0 then
```

Comment:
Looks like a scope issue there (I'm not even sure the local is limited to the `else` and new_pge is nil after the `end` - but it feels it would be the case).